### PR TITLE
rocksdb: fix rocksdb build under mingw64

### PR DESCRIPTION
**Pull request by yihuang**
###### Description of changes

```
$ nix-build -A pkgsCross.mingwW64.rocksdb
$ nix-build -A pkgsCross.mingwW64.rocksdb.tools
```

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/rocksdb/default.nix [hunk 1]
```diff
@@ -63,10 +67,13 @@ stdenv.mkDerivation rec {
   # otherwise "cc1: error: -Wformat-security ignored without -Wformat [-Werror=format-security]"
   hardeningDisable = lib.optional stdenv.hostPlatform.isWindows "format";
 
-  preInstall = ''
+  preInstall = (if stdenv.hostPlatform.isWindows then ''
+    mkdir -p $tools/bin
+    cp tools/{ldb,sst_dump}.exe $tools/bin/
```
Comment (wegank) [inline pkgs/development/libraries/rocksdb/default.nix:72]:
```suggestion
  preInstall = ''
    mkdir -p $tools/bin
    cp tools/{ldb,sst_dump}${stdenv.buildPlatform.extensions.executable} $tools/bin/
```

Comment (yihuang) [reply pkgs/development/libraries/rocksdb/default.nix]: done

**Comment by wegank** [comment]
Please squash the commits into one, with the commit message `rocksdb: fix build under mingw64`.

**Comment by yihuang** [comment]
> rocksdb: fix build under mingw64

done

**Comment by wegank** [comment]
Final check before merging: @ofborg build pkgsCross.mingwW64.rocksdb

**Review by wegank — APPROVED** [review]

File: pkgs/development/libraries/rocksdb/default.nix [hunk 2]
```diff
@@ -65,7 +69,7 @@ stdenv.mkDerivation rec {
 
   preInstall = ''
     mkdir -p $tools/bin
-    cp tools/{ldb,sst_dump} $tools/bin/
+    cp tools/{ldb,sst_dump}${stdenv.buildPlatform.extensions.executable} $tools/bin/
```
Comment (wegank) [inline pkgs/development/libraries/rocksdb/default.nix:72]: Sorry for the regression! This should be hostPlatform.

```suggestion
    cp tools/{ldb,sst_dump}${stdenv.hostPlatform.extensions.executable} $tools/bin/
```

**Review by wegank — APPROVED** [review]

**Pull request merged**